e2e/cypress/integration/multi_team_and_dm/multi_team_spec.js

```diff
@@ -147,4 +144,9 @@ describe('Send a DM', () => {
             cy.get('.badge').should('not.exist');
         });
     });
+
+    it('MM-T1805 No infinite loading spinner on Select Team page', () => {
+        cy.visit('/select_team');
```

Comment:
Should be fine to go directly but could you please satisfy the first 2 steps? Like create a team where the test user could join via UI at select team page, then visit again such page to check no more team to join and that the loading indicator does not exist. 
```
1. Create an account on a server with very few teams
2. Join all available teams teams
```

Reply:
Ah, I thought those steps are the 'arrangement' phase for this test, so should be done by using API.
Should I then move out my test from this suite? Because in the suite that arrangement of user and teams is done in `before`.

Reply:
We typically do test preparation directly via API whenever it's not the main focus of test, but prefer UI interaction especially when nothing yet exist to exercise such flow. Yeah, I agree to move into its own spec. Thanks @poflankov!

Reply:
@saturninoabril Thank you too for elaborate answer!
So, to make it clear, should I perform both steps 1 and 2 using UI?
And how should I name the new spec file?

Reply:
Step 1 via API then step 2 and the rest via UI. Name it related to test description (short, few words only :) )